### DISCUSSION TITLE
Discuss Identity features here!

### DISCUSSION THREAD

**Discussion by Draylar** [Draylar/identity#90] [General]
## Identity discussion page 🐔

Welcome to the general discussion page for the Identity repository. Feel free to leave not-entirely-issue posts or comments here.

**Comment by pigsteel** [comment]
identity pog???????

**Reply by Ausbin2** [reply]
I think its pretty pog, I love this mod :)

**Comment by pigsteel** [comment]
phantom movement should act like a constant velocity elytra! that'd make flying cooler

**Comment by notabirb1** [comment]
I think the UI should be changed. I don't like it covering the entire screen.

**Reply by Natedogwoooof** [reply]
thenn get ichuns instead

**Comment by DogsTongue** [comment]
Suggestion list(sorry for the inconvenience):

- Other players can ride players who are llamas or horses
- llamas ability is to shoot a ball of spit(like they do when hit in vanilla)
- config to allow the creeper explosion to hurt the player using it
- baby variants
- skeleton identity can ride spiders(including players using the spider identity

sorry for the long obnoxious list, I understand if you don't plan on implementing most or any of them.

**Reply by Draylar** [reply]
Just implemented your awesome Llama idea in 1.11.6. Thanks for the suggestions!

**Reply by MinecraftPerson** [reply]
You should make the Warden shoot a sonic boom, like the actual Warden, and possibly detect movement.

**Reply by Natedogwoooof** [reply]
> 
`` I would love that make it like that illagers handclaps whatever you call them I forgot what they're called make it like the Llama spit and the cow drinking milk like not like it but like make it the same keyblind as that because it just doesn't feel like you're the warden yet also add blindness but make it changeable in the config

**Reply by CombatCat74** [reply]
About the baby variants, how would you add a baby variant to the mod? I've been looking around your code for weeks and I cant figure out how to add a new mob.

**Comment by RageMage117** [comment]
Configuration for creature abilities would be nice. Example: You click on a mob and another menu comes up. That menu includes all abilities which you could turn on or off respectively. Upon changing one of these, it's possible that it may not be applied immediately and would therefore require to prompt the player to restart minecraft after making the desired changes. In the case of a server, the host would be informed to restart the server to apply the changes in the same way.

**Comment by testtenz** [comment]
how do i get players disguise?

**Reply by Draylar** [reply]
The public release version of Identity doesn't support disguising as other players.

**Reply by Ch1216** [reply]
Public? Is there a private release? If yes is there a way to get it?

**Reply by Draylar** [reply]
It's private because it was on commission (won't be released to the public).

I wouldn't add it to the main mod anyways; it doesn't fit my intended theme and would be a pain to maintain due to a few fun bugs. If anything, it would be a separate extension mod... maybe I'll recreate it some time down the road?

**Comment by CraftingKahuka** [comment]
one good idea is wolf should be able to eat rotten flesh without ill effect like the actual wolf mobs do since the wolf really doesn't have any abilities anyway it would be a good fit

**Reply by Draylar** [reply]
Wolves are immune to the effects of Rotten Flesh, Raw Chicken, and Pufferfish in the next release. Good idea!

**Comment by Ch1216** [comment]
Command:
/identity grant @s everything
(Would grant every morph to a player, Useful for survival)

**Reply by James103** [reply]
And a similar command:
`/identity revoke @s everything`
(Would revoke every morph from a player, useful if you want to start over with your mob collection)

**Reply by MimzyFang** [reply]
i think `*` would be more fitting then `everything`

**Reply by Draylar** [reply]
Added in the next release!

**Comment by Ch1216** [comment]
So you know how creepers, Spiders and Enderman have special visions when you spectate them in spectator?, Well they do, My idea is to make a config to toggle having the special and not having the special visions, It could be seen a challenge or just something to be more immersive

**Comment by MimzyFang** [comment]
the ability to tame disguised players

**Reply by MimzyFang** [reply]
and if this makes it in it would be fun to have an option to disable the Identity menu for tamed disguised players

**Comment by Ch1216** [comment]
1. Idea for dolphins:
Have permanent Dolphins Grace and by pressing R you can give it to the nearest player + The extra breathing/suffocating time dolphin have (Check the Minecraft wiki)
2. Make wither skeleton by able to hold stone swords and still be able to give the Wither effect and Same for Stray but with Bow and slowness
3. Have variables be accessible by middle clicking on the mob then scrolling/clicking to select one

**Reply by Natedogwoooof** [reply]
Third was added but we are  missing horse Lama and Trader Lama Rabbit Villager and Tropical fish varents

**Comment by Nocch11** [comment]
Hello, can someone tell me how to immediately change the mob that i kiled? With command or plugin or something else
Thanks :)

**Reply by MimzyFang** [reply]
go into your .minecraft/config folder open the identity.json5 in a text editor and change forceChangeAlways to true

**Comment by minecraftedleen** [comment]
Make undead mobs unable to swim

**Reply by Natedogwoooof** [reply]
No

**Comment by fayeispro** [comment]
This: https://github.com/Draylar/identity/discussions/176

**Comment by MimzyFang** [comment]
Enable Spectator vision shades for mobs like Creeper, Spider, Enderman,...

**Reply by Natedogwoooof** [reply]
yes but make it an option in config

**Comment by ghost** [comment]
Hey when i use identitys that are able to fly like wither or phantom it cant fly can u make it able to fly

**Comment by WoofTubeYT** [comment]
how do you set it so only op players can change identity on server

**Comment by kTombTV** [comment]
Could you make it so that the Ender Dragon destroys blocks it flies into? Could enable or disable this feature if wanted to.

**Reply by Natedogwoooof** [reply]
Two things it might be destructive is so probably no but make it just like it's dragging breath and then there

**Comment by GabiBrawl** [comment]
How can I edit the mod? I wanted to add block identities, but when I open a file of a ready identity of a mob to use as a draft, I only see [encrypted?] numbers.

And, could you add the block identities? Thx XD

**Comment by Spiderfffun** [comment]
When's 1.18 coming? I wanted to create a server with the mod, and i think it wouldn't be that great to add it later.

**Reply by Draylar** [reply]
No ETA, but I will be using the new Architectury setup so I can develop the Fabric+Forge versions at the same time. This will require a tad bit of a rewrite for the Fabric end as well as a lot of bug fixing. I hope to get an alpha out soon.

**Reply by Natedogwoooof** [reply]
> No ETA, but I will be using the new Architectury setup so I can develop the Fabric+Forge versions at the same time. This will require a tad bit of a rewrite for the Fabric end as well as a lot of bug fixing. I hope to get an alpha out soon.

Question if the player is Warden in the deep dark I think a warden shouldn't spawn or if it does the it shouldn't attack the player if they're a warden

**Reply by James103** [reply]
In other words, when a Sculk Shrieker checks for nearby Wardens, it also should search for players disguised as a Warden within 24 m of itself. It if finds any, the activation should fail.

**Comment by I-p5a** [comment]
Could you make the invisibility effect make the entity you're disguised as actually invisible?

**Comment by I-p5a** [comment]
it would be awesome if the entity you're disguised as supports the resource pack's animated model, like Fresh Animations

**Reply by WindowsXPFake** [reply]
I believe it does support fresh animations, because when I morph into a mob with Fresh Animations enabled it seems to work fine, but that could be cause I have the Entity Model Features Mod and Entity Texture Features Mod which allows Fresh Animations to work properly without optifine installed.

**Comment by UndergroundBeef** [comment]
I have a numerable suggestions, so I'm going to categorise them into groups of priority. I don't want anyone to think that this means I don't like the mod, it's one of my favourites of all time. All considerations would be appreciated

### Most Important:
These are features that I feel are 'missing' from the game
1. **Guardian lasers**
One of the most iconic features of guardians is the fact that their eyes shoot lasers. I was actually slightly surprised to find out that this was included in the mod. I have a simple solution to this: the ability icon will only appear if your crosshair is targeting a mob (otherwise it's semi-transparent). Players can hold down _r_ to shoot the laser. If the crosshair stops targeting the mob or the player stops holding down _r_, the laser will reset. As with vanilla mechanics, the laser charges up until it's yellow, then deals damage to the player
2. **Shulker bullets**
This is another feature that I was confused wasn't already in the game. Similar to the guardian laser in the previous suggestion, the ability icon for firing a shulker bullet will usually be semi-transparent. It will become fully visible whenever there is a mob within range. Whichever mob is closest to the crosshair will be the target. The shulker bullet then acts as normal, dodging and weaving until it hits the player. This ability would have 5-10 second cooldown
### Highly Reccomended:
These features would help increase the relation and similarities between the mod and the game itself
1. **Wither Phase 2**
 In vanilla Minecraft, withers gain armour when below 50% health. This armour stops ranged projectiles from dealing damage to them. This would be a great addition to the wither identity. I think this would improve the withers, because currently they don't seem very special. I feel like bosses should be noticeable better than the other mobs, whereas right now the wither doesn't feel out of the ordinary. This corresponds very well with the current wither boss
2. **Strays apply slowness**
Most of the mobs should have their own unique perks, therefore increasing the variety of choice between mobs. A quite simple example of this would be that strays could apply slowness to entities they damage. After all, this is the main purpose of strays being added to the game
3. **Guardians deal thorns damage**
This is another defining feature of guardians in minecraft. When any mob hurts a guardian, the guardian will deal thorns damage back at the attacker.
4. **Zombified piglin swarming**
Given part of the mod is the fact that individual mobs have unique perks, I feel like this is a slight missed opportunity. The idea would be that when something attacks you while you're in zombified piglin form, all zombified piglins in the direct area come to your aid, just like with the naturally spawning ones. This therefore means that the player benefits from this identity, as is the purpose of killing any mob. If this is to op, there could be a limit on how many of them come to your aid, or maybe each one only deals 1 hit
5. **Boss mobs destroy blocks**
This is a feature that I feel would be well earned after defeating a boss. Dragons always destroy the blocks they come in contact with, unless the block falls under the tag _#dragon\_immune_. Withers destroy the immediate blocks around them in a cube whenever they take damage, unless these blocks are _#wither\_immune_. I would not call these too op, after all players have to defeat a final boss to unlock this power
6. **Vexes phase through blocks**
This would be another link back to the original game. One of the most annoying things about vexes is the fact that they can pass through blocks to hit the player. I think that this is a defining trait of the vex, and should not be missed out. I understand that this would be very over-powered ability, and vexes can be quick to kill. Therefore, I think that this ability should have a 30 second cooldown at least, and can only be active for ~15 seconds. To avoid suffocation, the block that the player is in will be turned to air when the time runs out. If the player is inside a non-breakable block such as an end portal frame or bedrock, they will be teleported to the closest available block and the air pocket will be placed there
### Playability Suggestions
These suggestions are much more optional that the other ones
1. **Rapid projectile succession**
This applies to mobs like blazes and withers, but not ghasts or dragons. The idea is that the player can shoot multiple projectiles in rapid succession before the cooldown, much like the base mobs can. Blazes, for example, shoot 3 fireballs each time. Withers also shoot wither skulls in short rapid bursts. I had an idea on how this feature could be implemented into the mod. A small number, similar to item stacking, will appear next to the ability icon. Blazes will have a small number 3 next to theirs and withers will have a small 2. For a bonus, the blazes could light on fire when their ability is charged. When fighting blazes in vanilla, they light on fire as a warning that they are about to shoot, which I think matches my idea well. The cooldown after shooting all 3 fireballs would have to be made slightly longer, but I think the wither cooldown should stay the same. Out of the 3 playability suggestions, I would say this is the most plausible by far
2. **Information regarding identities**
This would be a great way to inform new players who don't quite understand how some of the mobs work. By right-clicking an identity in the menu (or maybe shift+clicking), an infobox appears with the mobs's base stats and abilities. The base stats would be Max Health (player default is 20), attack damage (player default would be 1) and movement speed (player default would be 100% for easy reference and comparison). It would then explain the passive abilities (such as iron golems inflicting knockback with fists, bees inflicting poison, wither skeletons shooting fire arrows and applying wither effect, striders walking on lava, fish breathing underwater, etc). Underneath that, it would list the special abilities (the ones that you press _r_ to activate). Underneath that, it would show weaknesses (such as blazes and endermen not being able to touch water). I just think this would help all players get a better understanding of the identities
3. **Identity menus**
My final suggestion is to add a keybind that opens up the identity menu. This lets you customise the mob you've morphed into. You could change your cat variant, modify the size of your slime, change your villager proffession (currently cosmetic), or dye your sheep (requires dye sacrifice).
 
This list is already getting quite long, so I think I'll wrap it up here. These are all of my suggestions for the game. Some of them would just be cool, whereas some of them I feel more strongly about. Overall, it's a great and I've been loving playing it. Hope you consider

**Comment by Shadowysn** [comment]
I tried a MC-Bedrock Morph mod and one of the small features they had is:
- When your Morph approaches a mob that they're scared of, the screen shakes. The closer you are, the more violent the shaking gets.

An unpleasant effect that discourages Identity-morphed players from being in vicinity of the Identity's feared objects sounds like a good addition.

**Comment by foics** [comment]
Animations for fox sitting and sleeping when your morphed as it. And any other mobs that have animations.

**Reply by foics** [reply]
also when using the wolf or fox identities strength potions or any sort of damage modification does not work other than weapons

**Comment by Ausbin2** [comment]
Can some mobs not fly? I tried bee and bat and allay ender dragon but none of them seem to be able to fly in survival mode which is the mode I intend on playing. Perhaps im misunderstanding the controls but I cant seem to make it work

**Reply by LucidREM** [reply]
i did not have any issues with flight. i changed to the bee and double tap jump just as in creative mode flight.

**Reply by CreeprVictor** [reply]
Mabey its because you switched to them in creative, then went in survival, 
I've experienced that bug many times

**Comment by LucidREM** [comment]
Can you alter the hitbox coding. While using Origins and Pehkui changing size does NOT change the hitbox of the chosen disguise. It is not a huge issue but does complicate gameplay when smaller or larger than default size.

**Comment by UndergroundBeef** [comment]
Around 6 months ago, I wrote my opinions and feedback on the mod (https://github.com/Draylar/identity/discussions/90#discussioncomment-2181390). I have recently realised that what I wrote was much too long, and didn't need to cover as much reasoning as it did. Here are my new and revised ideas:

**1. Guardian Lasers**
Hold the ability key and point your crosshair at a mob use the laser. The laser charges up and deals damage like it does with normal guardians, and shifting your crosshair off the mob resets it as it does normally. There would be a very short cooldown, because the attack still takes time to charge (maybe the small ability icon in the bottom right could be translucent when you're not targeting a mob)

**2. Shulker Bullets**
This ability also works like it does with vanilla shulkers. When you crosshair is targeting a mob, you press the ability key to fire the shulker bullet. This bullet pathfinds to the entity like normal. If there's no entity selected, maybe it just pathfinds to the targeted block (maybe). This would also make finding an endcity seem much more rewarding

**3. Strays Apply Slowness**
A very basic feature of the game is that strays apply slowness to the arrows they shoot. Please add this

**4. Guardians Deal Thorns Damage**
One of the defining traits of guardians is that they deal thorns damage. This would be another great reason to make finding a sea temple worth it

**5. Buffed Wither**
In the current version of the mod, there is little reward for killing the wither. Although the wither skull attack does good damage, it has an incredibly long cooldown, and does less overall dps than a blaze. The two things that would help even make the wither fight more worth it would be the wither shield, and possibly a reduced cooldown on the wither skull attack. The wither shield acts as it does in vanilla, blocking all projectile attacks when your health drops below 50%. I also appreciate that the cooldown of abilities can be modified in the config file, but most new players don't know how to access it

**Comment by Natedogwoooof** [comment]
add llama variations and give dolphen dolphens grace effect

**Comment by kopaka04** [comment]
Would it be possible to create a gamerule/command/config which allows/denies certain players to morph/open the morph menu? This would be really good with the origin mod as a "shapeshifter origin"
Or going further there could be a specified list of what mobs can a player morph into. For example: /something playername allow/deny onemob/allmobs
By allowing certain players to morph into certain mobs it would be possible to add new abilities to the origins (the blazeborn could only morph into blaze, magmacube, the fantom could have the undead creatures and etc...

This feature could bring more variety (and new possibilities to roleplay servers) to the use of the identity mod.

**Comment by Ptimou** [comment]
The /identity equip & /identity grant commands should be edited so that they can also be provided with a @e[limit=1] selector, to paste in the resulting identity the data from an existing mob.

**Comment by CombatCat74** [comment]
About the baby variants, how would you add a baby variant to the mod? I've been looking around your code for weeks and I cant figure out how to add a new mob.

**Comment by Papyrusthepuzzlemaster** [comment]
I'd like it if the Enderman was able to teleport on top of the block they're looking at rather than just teleporting to the sides and falling

**Comment by ItsFoxy87** [comment]
A feature where if you crouch (or hit R) as a fox, you can sleep :3

**Comment by CreeprVictor** [comment]
I think I have some great ideas for the Identity mod!
please read this, I put a lot of time, thought, and effort into this. If you see this, it would be greatly appreciated for you to read this, I hope the creator sees this!

Selecting Mobs & Missing mob variants: I've noticed I'm not able to select some mob variants, such as the other rabbit types, mabey, to reduce clutter and solve this issue, you could add like a drop down thing for some mobs to select different variants of the mob! (Like the creative inventory in BE). And for more complex mob variations, like horses and tropical fish (Or have this be the default as opposed to the dropdown), you'd have a GUI to where you can chose/edit the variant! Where you would have to kill the mob with the variant to get the variant! So, if you killed a black horse with spots, you can only become that, kill a white one with no decal, you could now be like a black horse with no decal, and vise versa!

Mob Abilities: Some mobs are missing some abilities, such as guardian lasers, evoker spawning vexes, vexes going through walls, etc.  Now, I know, there is only 1 ability button, but, that is where # 3 comes in!

Food: Some morphs should be able to eat their foods, like cows eat wheat, but not meat, etc.

Adding More Ability Keys, using 1-9 keys: Instead of being restricted to just one, we use the number keys 1-9! While morphed into a mob, you instead use the number keys for abilities! (Talked about more under "Possibilities" and under "Abilities"). And some abilities would require you to hold the ability button (Like creeper and blaze fireballs)! And they could be shown in a 2nd hotbar above the original!

Vision Shaders: When morphed into a mob like a creeper, enderman, or spider, you get the shader applied like when you spectate them!  (mabey even add some to other mobs to show how their real-world counterparts see!)

Possibilities
With all the stuff here, you would really more so be able to really get the experience of morphing into another mob!

Now we can add more abilities, like:

Evoker: 1= Evocation fangs 2= Vexes 3(mabey)= Convert (Blue sheep to red)

Blaze: 1= Shoot fireballs 2= toggle float down/hover/float up

Etc, more is shown under "Abilities"

Abilities
Axolotl: Regeneration
Bee: Pollinate
Blaze: Shoot 5 fireballs instead of 1
Blaze: toggle float down/hover/float up
Camel: Dash (by holding jump)
Chicken falls slowly
Chicken: Lay egg
Creeper: Explode(which you have)
Creeper has a creeper's vision
Ender Dragon: Shoot dragon breath ball (which you have)
Ender Dragon: Breath dragon breath
Enderman: Teleport (which you have)
Enderman: Random teleport (like chorus fruit)
Enderman: Ender pearl
Enderman has inverted vision
Evoker: Fangs (which you have)
Evoker: Vexes
Evoker(mabey): Convert(Blue sheep to red)
Fox: Dive
Ghast: Shoot fireball (which you have)
Goat: Ram
Goat(Mabey): Precision Jump (hold it, and it will show you where you can jump to, release, and you jump there! This would work similar to in the goat's AI)
Guardian (&Elder): Guardian laser
Guardian (&Elder) have thorns effect (which you have)
Hoglin zombifies
Horse, donkey, & mule: hold jump for jump like when riding a horse
Illusioner: Illusion
Panda (Playful): Roll
Piglin zombifies
Pufferfish: Puff up
Ravager: Roar
Ravager breaks leaves and crops
Sheep: Eat grass, grow wool
Sheep is shearable
Shulker is stuck in a fixed position aligned with blocks
Shulker can be on the sides and bottom of blocks
Shulker: Teleport
Shulker: Teleport Random
Shulker: Open/Close Shell
Shulker: Shoot shulker bullet (only when shell is open)
Silverfish: Hide in stone
Slime and magma cube split on death (mabey)
Sniffer: Dig
Snow golem: Shoot (only 1) snowball
Snow golem leaves snow trail
Spider climbs walls (which you have)
Spider doesn't get slowed by cobwebs
Spider has spider vision
Squid (&Glow squid): Ink jet
Strider walks on lava (which you have)
Strider gets cold out of lava
Tropical fish can be edited to be any of the [3,072 different unique combinations by choosing any of 2 shapes, 6 patterns, 16 base colors, and 16 pattern colors. Also, having presets of the 22 preset designs](https://minecraft.wiki/w/Tropical_Fish#Varieties)
Vex: Aggro (shows the red attacking skin)
Vex can go through blocks (and is forced into fly mode)
Warden: Sonic Boom (which you have)
Warden (mabey): Sniff (shows nearby mobs, including players)
Warden (mabey) sees vibrations
Warden (mabey) uses the "Sobel" shader that was featured in the "Super Secret Settings" a while back. ([link to a mod that adds this in newer versions](https://www.curseforge.com/minecraft/mc-mods/super-secret-settings))
Witch: Drink potion
Witch: Throw negative potion
Witch: Throw helpful potion
Wither: Shoot wither skull (which you have)
Wither: Shoot blue wither skull
Wither: Toggle shield
Wither is unaccessable in survival mode without commands (can be bypassed by going into creative mode, morphing, then survival)
Wolf: Aggro (shows aggressive skin)
Other and bugs
Other things I've noticed include:

Silverfish morph can't hide in stone
Shulker morph not restricted to block alignment
Shulker morph can walk
Piglin don't know if to attack you or not when you are a piglin and aren't wearing gold, they freak out.
Piglin also freak out whenever you are morphed not wearing gold for the same reason
Piglin and hoglin morphs don't convert to zombies
Fox morph can't dive
Goat morph can't charge
Blaze morph only shoots 1 fireball
Horse, donkey, and mule morph has normal player speed, jump, and can't auto step up 1 block
Horse, panda and rabbit are missing their variants
No brown mooshroom
Bee morph can't pollinate
Dolphin morph dies shortly on land
Dolphin morph can't ever drown
Axolotl morph never dries out
Guardian and elder guardian morphs die on land
Phantoms aren't scared of cat or ocelot morph
Creepers aren't scared of cat morph and aren't scared of ocelot morph in creative mode
Foxes are scared of morphed players
among other morphs not being scared of morphed they should be
Sniffer, axolotl on land, and other morphs have player speed
Chicken doesn't have slow falling
Ravager morph doesn't break leaves and crops
Spider morphs get stuck in cobwebs
Creeper morph explosion doesn't damage blocks and you can't defuse yourself (I also think you should take some damage, or atleast be affected by the explosion, dealing you fall damage)
Endermite and silverfish sink all the way through 1 soul sand block
Foxes don't attack chicken morph
Polar bear doesn't attack fox morph
I could be wrong, but I don't think witches heal illager morphs in raids
Wither skeleton doesn't attack piglin morph
Piglin groups don't attack wither skeleton morph
Strider morph warm even when not on lava
Ravager can't roar
Vex takes fall damage

**Reply by Natedogwoooof** [reply]
im sorry to say but this mod is kina dead

**Reply by CreeprVictor** [reply]
Still does seem to be the most popular up-to-date morphing mod

**Reply by Natedogwoooof** [reply]
dead is in not getting much updates.

**Reply by Natedogwoooof** [reply]
Good news they added the feature you wanted to remorphed

**Reply by CreeprVictor** [reply]
how many of them? And which one(s)?

**Reply by Natedogwoooof** [reply]
They added the head thing on menu so you don't need to press a button to become player anymore

**Comment by WindowsXPFake** [comment]
Any chance that morph NBT Data can be implemented? so like if you kill a villager with any profession when you morph into the acquired morph that morph will have the same profession as the villager you killed

**Comment by EnchantedBread** [comment]
Im having issues with my config, it seems nothing I change has any effect on the game, I want to block flying behind some advancements but cant find any documentation that tells me how to format it. did I do it wrong and that's why my entire config is being ignore (on sever & client)

  //A list of Advancements required before the player can fly using an Identity.
  "advancementsRequiredForFlight": [nether/get_wither_skull, nether/find_fortress, nether/obtain_ancient_debris],

...

  //An override map for requiredKillsForIdentity for specific entity types.
  "requiredKillsByType": {
    "minecraft:elder_guardian": 1,
    "minecraft:wither": 1,
    "minecraft:ender_dragon": 1,
    "minecraft:bee": 1,
    "minecraft:spider: 10,
    "minecraft:zombie: 10,
    "minecraft:skeleton: 10,
    "minecraft:cow: 10,
    "minecraft:pig: 10,
    "minecraft:sheep: 10,
    "minecraft:creeper: 10
  },
  "abilityCooldownMap": {
    "minecraft:blaze": 20,
    "minecraft:enderman": 100,
    "minecraft:witch": 200,
    "minecraft:creeper": 100,
    "minecraft:wither": 200,
    "minecraft:ghast": 60,
    "minecraft:ender_dragon": 20,
    "minecraft:snow_golem": 10,
    "minecraft:evoker": 10
  }
}

**Reply by EnchantedBread** [reply]
[identity.json](https://github.com/user-attachments/files/16260652/identity.json)


(renamed from .jason5 for upload)

**Comment by tobisworstnightmare-hash** [comment]
Can you please add support for simple modded flying entities (such as with a tag) I'm a modder and can't really figure it out.
It would also be helpful to have tags for other entity traits.
